sklearn/ensemble/_forest.py

```diff
@@ -505,7 +522,72 @@ def __init__(self,
             verbose=verbose,
             warm_start=warm_start,
             class_weight=class_weight,
-            max_samples=max_samples)
+            max_samples=max_samples,
+            importance_type=importance_type)
+
+    def _set_oob_permutation_importance(self, X, y):
```

Comment:
So I assume that this function could be moved into the `BaseForest` and could accept a `scorer` parameter.